common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -183,53 +180,15 @@ static byte getByte(long address) {
     }
 
     static short getShort(long address) {
-        if (UNALIGNED) {
-            return UNSAFE.getShort(address);
-        } else if (BIG_ENDIAN) {
-            return (short) (getByte(address) << 8 | getByte(address + 1) & 0xff);
-        } else {
-            return (short) (getByte(address + 1) << 8 | getByte(address) & 0xff);
-        }
+        return UNSAFE.getShort(address);
     }
 
     static int getInt(long address) {
-        if (UNALIGNED) {
-            return UNSAFE.getInt(address);
-        } else if (BIG_ENDIAN) {
-            return getByte(address) << 24 |
-                  (getByte(address + 1) & 0xff) << 16 |
-                  (getByte(address + 2) & 0xff) <<  8 |
-                   getByte(address + 3) & 0xff;
-        } else {
-            return getByte(address + 3) << 24 |
-                  (getByte(address + 2) & 0xff) << 16 |
-                  (getByte(address + 1) & 0xff) <<  8 |
-                   getByte(address) & 0xff;
-        }
+        return UNSAFE.getInt(address);
     }
 
     static long getLong(long address) {
-        if (UNALIGNED) {
```

Comment:
release notes should call out that people who cheat and use internal packages should be ready for some pain :wink: